### ISSUE TITLE
Spotlight action onTrigger execution order incorrect when using keyboard

### ISSUE THREAD

**Issue by leonardomathon** [mantinedev/mantine#986]
### What package has an issue

@mantine/core

### Describe the bug

This bug occurs when you create a spotlight provider without any actions and register the actions later using a useEffect hook with registered state (so that it only executes once). 

Suppose this actions array, has a special action with id `action2` for which the trigger changes based on some state (textInput state in my example). Therefore, whenever this state changes, we deregister the action and register it again with a different function in the onTrigger. 

However, we want to have the special action at the second place of the action list so we use a group. This works well. If we now click on all the actions, the output is correct but when we use the arrow keys and hit enter on a selected action, it seems that the order of the actions has changed. The special action is at the end of the actions array but is still rendered at the second place from the top. If we search for the action, and then hit enter, the right output is shown.

I have created a minimal example (see codesandbox link below):
You can open the spotlight using "mod+K". If you click on all the items one by one it will output the following in the console:
```
Action 1 
Special Action 2 
Action 3 
Action 4 
```
If you use the arrow keys on the keyboard, the output will be:
```
Action 1 
Action 3 
Action 4 
Special Action 2 
```
If you first make a search and then use the arrows, the output is correct.

For clarity, here is a GIF showing the problem (watch that the output changes):
![mantine](https://user-images.githubusercontent.com/30813241/157840965-502823c0-d6e0-475d-86fd-7bc3cb92838c.gif)

Unfortunately, I am too busy with graduation and work so I won't be able to help solve this issue. If anything is unclear, feel free to reach out to me!

### In which browser did the problem occur

Chrome v99.0.4844.51

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/green-smoke-zcdd0u?file=/src/ExampleComponent.tsx

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

No

**Comment by tsar-boomba** [comment]
I'll look into this bug for you and work on a fix. Looks like the grouping isn't updated when actions are updated.

**Comment by rtivital** [comment]
Fixed in 4.0.3
